dashboard/modules/log/log_agent.py

```diff
@@ -394,7 +394,10 @@ def _resolve_filename(cls, root_log_dir: Path, filename: str) -> Path:
         else:
             filepath = Path(filename)
 
-        filepath = filepath.resolve()
+        # We want to allow relative paths that include symlinks pointing outside of the
+        # `root_log_dir`, so use `os.path.abspath` instead of `Path.resolve()` because
+        # `os.path.abspath` does not resolve symlinks.
+        filepath = Path(os.path.abspath(filepath))
```

Comment:
Should we use this https://docs.python.org/3/library/pathlib.html#pathlib.Path.absolute ?

<img width="902" alt="image" src="https://github.com/ray-project/ray/assets/11676094/e228b3d4-bc6a-4392-aa00-27a42cf75143">

Reply:
We do want to normalize the resulting path here to remove any `../`s (for example the `# Test relative path recursive up is not allowed` test case would fail if we were to replace with this)